dist/api_data.py

```diff
@@ -1369,6 +1369,11 @@ def __ge__(self, other):
     Config('background', '', r'''
         enable/disabled the background compaction server.''',
         type='boolean'),
+    Config('exclude', '', r'''
+        If non-empty and used when enabling background compaction, exclude the table object targets
```

Comment:
"and used when enabling background compaction" -- do you mean to say that this list can only be provided when enabling the background compaction? Maybe a little re-wording, eg:

"A list of table objects to be excluded from background compaction. The list is immutable and only applied when the background compaction gets enabled. The list is not saved between the calls and needs to be reapplied each time the service is enabled. The individual objects in the list can only be of the "table:" URI type."

Reply:
The testing should include the API requirements here:
1. List only applicable with service enable, not with disable.
2. If enable called again with a different list, what happens?
3. If enable called again with the same list, what happens?
4. If disable called with a list, it should be an error
5. error if individual items in the list are not table type
6. What happens if a table in the list doesn't exist yet?
7. What happens if a table gets deleted later (and then recreated possibly? - Is MongoDB safe from using safe wiredtiger table namespace in this case?)
8. What happens if two competing calls are made to `session.compact(background=..., ...)` . How do we make sure only one succeeds.

Reply:
1. See [here](https://github.com/wiredtiger/wiredtiger/pull/9719/files#diff-eab825f891ddcec54bad4fdc922828201330ef985f8458558ee6e20615658a75R416-R420)
2. Background compaction has to be disabled before being enabled again, see [here](https://github.com/wiredtiger/wiredtiger/blob/develop/src/conn/conn_compact.c#L588-L593).
3. Ditto
4. Same as 1
5. See [here](https://github.com/wiredtiger/wiredtiger/pull/9719/files#diff-ad85b77b14594b779f82d482423c144c3287a45c607d76e7a68db8466ac93e30R705-R708).
6. It is ok, we don't check for their existence.
7. Once a name is set in the excluded list, it will excluded as long as the background compaction is enabled. Whether the table is created later on, deleted, recreated again, etc, the name is blacklisted. I believe this could be an issue with MongoDB if they create the table "a" that ends up being "collection-123" at the WiredTiger level, then delete "a" and create "b" that is also generated with the same id since "collection-123". Is it possible to generate the same collection id though? cc: @kommiharibabu - something we discussed earlier.
8. We have a lock when interacting with the background compaction server, see [here](https://github.com/wiredtiger/wiredtiger/blob/develop/src/conn/conn_compact.c#L579).

Reply:
For point 7 - I am pretty sure the collection names are not repeated. I don't think we have to worry about it. If they are it is for the WiredTiger's application to correct that, would not require a WiredTiger change.